lib/scholar/stats.ex

```diff
@@ -0,0 +1,184 @@
+defmodule Scholar.Stats do
+  @moduledoc """
+  Statistical functions
+
+  All of the functions in this module are implemented as
+  numerical functions and can be JIT or AOT compiled with
+  any supported `Nx` compiler.
+  """
+
+  import Nx.Defn
+
+  general = [
+    axis: [
+      type: {:or, [:non_neg_integer, :atom]},
+      default: 0,
+      doc: """
+      Axis to calculate the operation. If set to `nil` then
+      the operation is performed on the whole tensor.
+      """
+    ],
+    keep_axis: [
+      type: :boolean,
+      default: false,
+      doc: "If set to true, the axis which is reduced is left."
+    ]
+  ]
+
+  moment_schema =
+    general ++
+      [
+        moment: [
+          type: :pos_integer,
+          default: 1,
+          doc: "Order of central moment that is returned."
+        ]
+      ]
+
+  skew_schema =
+    general ++
+      [
+        bias: [
+          type: :boolean,
+          default: true,
+          doc: "If false, then the calculations are corrected for statistical bias."
+        ]
+      ]
+
+  kurtosis_schema =
+    general ++
+      [
+        bias: [
+          type: :boolean,
+          default: true,
+          doc: "If false, then the calculations are corrected for statistical bias."
+        ],
+        variant: [
+          type: {:in, [:fisher, :pearson]},
+          default: :fisher,
+          doc:
+            "If :fisher then Fisher's definition is used, if :pearson then Pearson's definition is used."
+        ]
+      ]
+
+  @moment_schema NimbleOptions.new!(moment_schema)
+  @skew_schema NimbleOptions.new!(skew_schema)
+  @kurtosis_schema NimbleOptions.new!(kurtosis_schema)
+
+  @doc """
+  Calculates the nth moment about the mean for a sample.
+
+  ## Options
+
+  #{NimbleOptions.docs(@moment_schema)}
+
+  ## Examples
+
+      iex> x = Nx.tensor([[3, 5, 3], [2, 6, 1], [9, 3, 2], [1, 6, 8]])
+      iex> Scholar.Stats.moment(x, moment: 2)
+      #Nx.Tensor<
+        f32[3]
+        [9.6875, 1.5, 7.25]
+      >
+  """
+  deftransform moment(tensor, opts \\ []) do
+    opts = NimbleOptions.validate!(opts, @moment_schema)
+
+    axis =
+      if axis_opt = opts[:axis] do
+        Nx.Shape.normalize_axis(Nx.shape(tensor), axis_opt, Nx.names(tensor))
+      end
+
+    opts = Keyword.put(opts, :axis, if(axis == nil, do: nil, else: [axis]))
+    num_samples = if opts[:axis] == nil, do: Nx.size(tensor), else: Nx.axis_size(tensor, 0)
+    moment_n(tensor, num_samples, opts)
+  end
+
+  defnp moment_n(tensor, num_samples, opts) do
+    mean = Nx.mean(tensor, axes: opts[:axis], keep_axes: opts[:keep_axis])
+
+    Nx.sum((tensor - mean) ** opts[:moment], axes: opts[:axis], keep_axes: opts[:keep_axis]) /
+      num_samples
+  end
```

Comment:
If moment is reuqired, maybe it should be a value instead of an option? Also, could we support `axes` and `keep_axes` instead? Something like this:

```suggestion
  deftransform moment(tensor, moment, opts \\ []) do
    opts = NimbleOptions.validate!(opts, @moment_schema)
    num_samples = Enum.sum(Enum.map(opts[:axes] || Nx.axes(tensor), &Nx.axis_size(tensor, &1)))
    moment_n(tensor, moment, num_samples, opts)
  end

  defnp moment_n(tensor, moment, num_samples, opts) do
    mean = Nx.mean(tensor, opts)
    Nx.sum((tensor - mean) ** moment, opts) / num_samples
  end
```

Reply:
Yeah, the number of moments as a value is a better idea. We can also add multiple axes, but then we will calculates moments of moments of moments and so on. I will add information about it in doc and it seems good to me.